packages/dev/core/src/Meshes/Builders/greasedLineBuilder.ts

```diff
@@ -143,40 +147,64 @@ export function CreateGreasedLine(name: string, options: GreasedLineMeshBuilderO
     scene = <Scene>(scene ?? EngineStore.LastCreatedScene);
 
     let instance;
-    const allPoints = GreasedLineMesh.ConvertPoints(options.points);
-
-    let length = 0;
-    if (Array.isArray(allPoints[0])) {
-        allPoints.forEach((points) => {
-            length += points.length / 3;
-        });
-    }
+    const allPoints = GreasedLineTools.ConvertPoints(options.points);
 
     options.widthDistribution = options.widthDistribution ?? GreasedLineMeshWidthDistribution.WIDTH_DISTRIBUTION_START;
+    if (options.ribbonOptions) {
+        options.ribbonOptions.facesMode = options.ribbonOptions.facesMode ?? GreasedLineRibbonFacesMode.FACES_MODE_SINGLE_SIDED_NO_BACKFACE_CULLING;
+        options.ribbonOptions.pointsMode = options.ribbonOptions.pointsMode ?? GreasedLineRibbonPointsMode.POINTS_MODE_POINTS;
+        options.ribbonOptions.directionsAutoMode =
+            options.ribbonOptions.directionsAutoMode ??
+            (options.ribbonOptions.directions ? GreasedLineRibbonAutoDirectionMode.AUTO_DIRECTIONS_NONE : GreasedLineRibbonAutoDirectionMode.AUTO_DIRECTIONS_FROM_FIRST_SEGMENT);
+    }
 
     materialOptions = materialOptions ?? {
-        color: GreasedLinePluginMaterial.DEFAULT_COLOR,
+        color: GreasedLineBaseMaterial.DEFAULT_COLOR,
     };
     materialOptions.createAndAssignMaterial = materialOptions.createAndAssignMaterial ?? true;
     materialOptions.colorDistribution = materialOptions?.colorDistribution ?? GreasedLineMeshColorDistribution.COLOR_DISTRIBUTION_START;
     materialOptions.materialType = materialOptions.materialType ?? GreasedLineMeshMaterialType.MATERIAL_TYPE_STANDARD;
 
+    let length = 0;
+    if (Array.isArray(allPoints[0])) {
+        allPoints.forEach((points) => {
+            length += points.length / 3;
+        });
+    }
+
     const widths = CompleteGreasedLineWidthTable(length, options.widths ?? [], options.widthDistribution);
 
+    if (options.ribbonOptions) {
+        if (options.ribbonOptions.pointsMode === GreasedLineRibbonPointsMode.POINTS_MODE_POINTS) {
+            // length *= 2;
+        }
+    }

```

Comment:
Remove this block, as it is doing nothing?

Reply:
removed